### PR TITLE
feat: [DCS-220] add count_usa_phone and percent_usa_phone functions

### PR DESCRIPTION
**Pull request by ashank2603**
## Description

Validity function for `count_usa_phone` and `percent_usa_phone`

## Type of change

Delete irrelevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [x] Locally Tested
- [ ] Needs Testing From Production

### CODE REVIEW

File: datachecks/core/validation/validity_validation.py [hunk 1]
```diff
@@ -48,3 +48,35 @@ def _generate_metric_value(self, **kwargs) -> Union[float, int]:
             raise NotImplementedError(
                 "UUID validation is only supported for SQL data sources"
             )
+
+
+class CountUSAPhoneValidation(Validation):
+    def _generate_metric_value(self, **kwargs) -> Union[float, int]:
+        if isinstance(self.data_source, SQLDataSource):
+            valid_count, total_count = self.data_source.query_string_pattern_validity(
```
Comment (WeryZebra-Yue) [inline datachecks/core/validation/validity_validation.py:56]: Check if we need total_count or you can remove it @ashank2603

File: datachecks/core/datasource/sql_datasource.py [hunk 2]
```diff
@@ -383,6 +383,7 @@ def query_get_duplicate_count(
 
     regex_pattern = {
```
Comment (WeryZebra-Yue) [inline datachecks/core/datasource/sql_datasource.py:384]: move this into the function @ashank2603

**Review by WeryZebra-Yue — CHANGES_REQUESTED** [review]

File: datachecks/core/datasource/sql_datasource.py (hunk 2 above)
Comment (WeryZebra-Yue) [reply datachecks/core/datasource/sql_datasource.py:384]: @ashank2603  Please create the same for SEARCH Datasource

**Comment by Ryuk-me** [comment]
Missing test cases for configuration parser @ashank2603 
please add it in `tests/core/configuration/test_configuration_v1.py`

**Review by subhankarb — APPROVED** [review]

**Pull request merged**